### PR TITLE
TestCase for Issue 52

### LINKED ISSUE #52: Issue with strpos call in laminas-stdlib when using IsFloat validator
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | latest

#### Summary

Initial misreported bug https://github.com/laminas/laminas-stdlib/issues/27

It seems when using this library in conjunction with laminas/laminas-stdlib:3.4.0, there is the potential for a boolean argument to be passed into the underlying wrappers `strpos` method, causing an exception to be thrown.

#### Current behavior

\Laminas\I18n\Validator\IsFloat::isValid:243 calls `strpos` and passes in $lastStringGroup which has the potential to be a boolean, causing an exception.

https://github.com/laminas/laminas-i18n/blob/b85a640138687f26129ace789fd24bbe0e0972ad/src/Validator/IsFloat.php#L240-L246

#### How to reproduce

Call \Laminas\I18n\Validator\IsFloat::isValid and pass in a string value (IE "1")

#### Expected behavior

\Laminas\I18n\Validator\IsFloat::isValid should not throw an exception

### PR DESCRIPTION
**Pull request by javabudd**
This adds a testcase for https://github.com/laminas/laminas-i18n/issues/52

|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

Adds a simple test case to address https://github.com/laminas/laminas-i18n/issues/52. This will need a `composer update laminas/laminas-stdlib` for tests to fail (need laminas/laminas-stdlib 3.4.0)

### CODE REVIEW

**Comment by javabudd** [comment]
Results after `composer update laminas/laminas-stdlib`:

```
There was 1 error:

1) LaminasTest\I18n\Validator\IsFloatTest::testStringValueReturnsTrue
TypeError: grapheme_strpos() expects parameter 1 to be string, bool given

/home/andyo/Documents/PhpstormProjects/laminas-i18n/vendor/laminas/laminas-stdlib/src/StringWrapper/Intl.php:87
/home/andyo/Documents/PhpstormProjects/laminas-i18n/src/Validator/IsFloat.php:243
/home/andyo/Documents/PhpstormProjects/laminas-i18n/test/Validator/IsFloatTest.php:183

--

There were 25 warnings:

1) LaminasTest\I18n\ModuleTest::testInitRegistersPluginManagerSpecificationWithServiceListener
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

2) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testFactoryReturnsUnconfiguredPluginManagerWhenNoOptionsPresent
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

3) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testCreateServiceReturnsUnconfiguredPluginManagerWhenNoOptionsPresent
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

4) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testFactoryCanConfigurePluginManagerViaOptions with data set #0 ('gettext')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

5) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testFactoryCanConfigurePluginManagerViaOptions with data set #1 ('getText')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

6) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testFactoryCanConfigurePluginManagerViaOptions with data set #2 ('GetText')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

7) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testFactoryCanConfigurePluginManagerViaOptions with data set #3 ('phparray')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

8) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testFactoryCanConfigurePluginManagerViaOptions with data set #4 ('phpArray')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

9) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testFactoryCanConfigurePluginManagerViaOptions with data set #5 ('PhpArray')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

10) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testCreateServiceCanConfigurePluginManagerViaOptions with data set #0 ('gettext')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

11) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testCreateServiceCanConfigurePluginManagerViaOptions with data set #1 ('getText')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

12) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testCreateServiceCanConfigurePluginManagerViaOptions with data set #2 ('GetText')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

13) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testCreateServiceCanConfigurePluginManagerViaOptions with data set #3 ('phparray')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

14) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testCreateServiceCanConfigurePluginManagerViaOptions with data set #4 ('phpArray')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

15) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testCreateServiceCanConfigurePluginManagerViaOptions with data set #5 ('PhpArray')
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

16) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testConfiguresTranslatorServicesWhenFound
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

17) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testDoesNotConfigureTranslatorServicesWhenServiceListenerPresent
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

18) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testDoesNotConfigureTranslatorServicesWhenConfigServiceNotPresent
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

19) LaminasTest\I18n\Translator\LoaderPluginManagerFactoryTest::testDoesNotConfigureTranslatorServicesWhenConfigServiceDoesNotContainTranslatorsConfig
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

20) LaminasTest\I18n\Translator\TranslatorServiceFactoryTest::testCreateServiceWithNoTranslatorKeyDefined
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

21) LaminasTest\I18n\Translator\TranslatorServiceFactoryTest::testCreateServiceWithNoTranslatorPluginManagerDefined
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

22) LaminasTest\I18n\View\Helper\TranslatePluralTest::testDefaultInvokeArguments
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

23) LaminasTest\I18n\View\Helper\TranslatePluralTest::testCustomInvokeArguments
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

24) LaminasTest\I18n\View\Helper\TranslateTest::testDefaultInvokeArguments
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

25) LaminasTest\I18n\View\Helper\TranslateTest::testCustomInvokeArguments
PHPUnit\Framework\TestCase::prophesize() is deprecated and will be removed in PHPUnit 10. Please use the trait provided by phpspec/prophecy-phpunit.

ERRORS!
Tests: 1789, Assertions: 19383, Errors: 1, Warnings: 25, Skipped: 1.

```

**Comment by froschdesign** [comment]
The following `if` condition must be updated to check if `$lastStringGroup` is a string:

https://github.com/laminas/laminas-i18n/blob/5e85a8facc5534e856cc7f5b4326533eede84b8a/src/Validator/IsFloat.php#L240-L244

**Comment by froschdesign** [comment]
Please use the [2.11.x](https://github.com/laminas/laminas-i18n/tree/2.11.x) branch for this bugfix.

**Comment by cvigorsICBF** [comment]
Thanks of for the effort, but the issue is persisting after I downgrade to 2.11.x
Downgraded laminas/laminas-stdlib to 3.3.1. Working again

File: test/Validator/IsFloatTest.php [hunk 1]
```diff
@@ -165,7 +165,7 @@ public function validationFailureProvider()
         $testingArray  = [
             'ar'    => ['10.1', '66notflot.6'],
             'ru'    => ['10.1', '66notflot.6', '2,000.00', '2 00'],
-            'en'    => ['10,1', '66notflot.6', '2.000,00', '2 000', '2,00'],
+            'en'    => ['10,1', '66notflot.6', '2.000,00', '2 000', '2,00', '1'],
```
Comment (arokettu) [inline test/Validator/IsFloatTest.php:168]: '1' should not be a failure, it's a valid float

File: src/Validator/IsFloat.php [hunk 2]
```diff
@@ -237,10 +237,10 @@ public function isValid($value)
         $groupSize = $formatter->getAttribute(NumberFormatter::GROUPING_SIZE) ?: 3;
         $lastStringGroup = $this->wrapper->substr($value, -$groupSize);
 
-        if ((preg_match($lnumSearch, $unGroupedValue)
+        if (false !== $lastStringGroup && false === $this->wrapper->strpos($lastStringGroup, $groupSeparator)
```
Comment (arokettu) [inline src/Validator/IsFloat.php:240]: I have a feeling it should be

`if ((false === $lastStringGroup || false === $this->wrapper->strpos($lastStringGroup, $groupSeparator))`

**Comment by javabudd** [comment]
Favoring https://github.com/laminas/laminas-i18n/pull/54

**Pull request closed without merging**